cmd/lookup/main.go

```diff
@@ -35,53 +46,66 @@ func main() {
 			continue
 		}
 
-		ref, err := registry.ParseRef(img)
-		if err != nil {
-			panic(err)
-		}
-
-		var obj any
-		if opts == zeroOpts {
-			// if we have no explicit type and didn't request a HEAD, invoke SynthesizeIndex instead of Lookup
-			obj, err = registry.SynthesizeIndex(ctx, ref)
-			if err != nil {
-				panic(err)
-			}
-		} else {
-			r, err := registry.Lookup(ctx, ref, &opts)
+		do := func(opts registry.LookupOptions) {
+			ref, err := registry.ParseRef(img)
 			if err != nil {
 				panic(err)
 			}
-			if r != nil {
-				desc := r.Descriptor()
-				if opts.Head {
-					obj = desc
-				} else {
-					b, err := io.ReadAll(r)
-					if err != nil {
-						r.Close()
-						panic(err)
-					}
-					if opts.Type == registry.LookupTypeManifest {
-						// if it was a manifest lookup, cast the byte slice to json.RawMessage so we get the actual JSON (not base64)
-						obj = json.RawMessage(b)
-					} else {
-						obj = b
-					}
-				}
-				err = r.Close()
+
+			var obj any
+			if opts == zeroOpts {
+				// if we have no explicit type and didn't request a HEAD, invoke SynthesizeIndex instead of Lookup
+				obj, err = registry.SynthesizeIndex(ctx, ref)
 				if err != nil {
 					panic(err)
 				}
 			} else {
-				obj = nil
+				r, err := registry.Lookup(ctx, ref, &opts)
+				if err != nil {
+					panic(err)
+				}
+				if r != nil {
+					desc := r.Descriptor()
+					if opts.Head {
+						obj = desc
+					} else {
+						b, err := io.ReadAll(r)
+						if err != nil {
+							r.Close()
+							panic(err)
+						}
+						if opts.Type == registry.LookupTypeManifest {
+							// if it was a manifest lookup, cast the byte slice to json.RawMessage so we get the actual JSON (not base64)
+							obj = json.RawMessage(b)
+						} else {
+							obj = b
+						}
+					}
+					err = r.Close()
+					if err != nil {
+						panic(err)
+					}
+				} else {
+					obj = nil
+				}
+			}
+
+			e := json.NewEncoder(os.Stdout)
+			e.SetIndent("", "\t")
+			if err := e.Encode(obj); err != nil {
+				panic(err)
 			}
 		}
 
-		e := json.NewEncoder(os.Stdout)
-		e.SetIndent("", "\t")
-		if err := e.Encode(obj); err != nil {
-			panic(err)
+		if parallel {
+			wg.Add(1)
+			go func(opts registry.LookupOptions) {
+				defer wg.Done()
+				// TODO synchronize output so that it still arrives in-order?  maybe the randomness is part of the charm?
+				do(opts)
+			}(opts)
```

Comment:
for loop variables and closures

Reply:
Yeah... making sure we copy things appropriately so we don't leak across invocations.